File: Python/decorator_pattern/cafe_adicional_de_leite.py
from cafe_decorator import CafeDecorator

class CafeAdicionalDeLeite(CafeDecorator):
    def __init__(self, cafe):
        super().__init__(cafe)
        
    def get_custo(self):
        custo = super().get_custo()
        custo = custo + 2.50
        return custo
    
    def get_descricao(self):
        descricao = super().get_descricao()
        descricao = f"{descricao} + Leite"
        return descricao

File: CSharp/FaculdadeVincit.DesignPatterns.DecoratorPattern/CafeAdicionalDeLeite.cs
namespace FaculdadeVincit.DesignPatterns.DecoratorPattern;

public class CafeAdicionalDeLeite : CafeDecorator
{
    public CafeAdicionalDeLeite(ICafe cafe) : base(cafe)
    {
    }

    public override string Descricao => $"{base.Descricao} + Leite";
    public override double Custo => base.Custo + 2.50;

}
